src/utils/flagCodes.ts

```diff
@@ -0,0 +1,195 @@
+// gasstation return example
+/*{
+  lastUpdate: '2020-03-03T11:23:53.500017Z',
+  lowest: '1000000001',
+  safeLow: '6000000001',
+  standard: '10000000001',
+  fast: '18800000001',
+  fastest: '1457792372351',
+} */
+// can in most cases safely override 9 last digits
+// even 10 digits if only using standard price
+
+export interface Flag<T extends string> {
+  name: T
+  values: string[] // value at 0-index is the default
+  meaningfulDigits: number
+}
+
+// OUR specific flags
+const SENTINEL = '12'
+
+type DxFlagName = 'provider' | 'mobile' | 'browser'
+
+// Providers currently detected by web3connect
+// new provider names can be appended to values as long as values.length - 1 <= meaningfulDigits
+const provider: Flag<DxFlagName> = {
+  name: 'provider',
+  values: [
+    'Web3', // generic name for undetected provider
```

Comment:
Web3connect detects providers unknown to it as Web3

Reply:
I don't know why, but I'm sure we'll get dependabot updating wallet connect one day, and braking some logics here

Ok, let's leave it like this then

Reply:
That's the beauty of appendable flags. As long as we don't overflow `meaningfulDigits` we can push to values array.

So any new provider names will get added to the flag.values, and if there ever a new default|unknown it will also be added and we'll see more of that unknown_prov and not Web3.

But we will still knowthat that unknown is essentially the same old Web3